arangod/Cluster/HeartbeatThread.cpp

```diff
@@ -1202,9 +1203,9 @@ void HeartbeatThread::runAgent() {
   // simple loop to post dead threads every hour, no other tasks today
   while (!isStopping()) {
     try {
-      CONDITION_LOCKER(locker, _condition);
+      std::unique_lock locker{_condition.mutex};
       if (!isStopping()) {
-        locker.wait(std::chrono::hours(1));
+        _condition.cv.wait_for(locker, std::chrono::hours{1});
```

Comment:
Looks really strange, why cannot we just not creating heartbeat thread on agency?
Anyway not related to this PR, just thoughts